runtime/metricsview/executor_wrap_clickhouse_compare_time_dim.go

```diff
@@ -0,0 +1,73 @@
+package metricsview
+
+import (
+	"fmt"
+	"hash/crc32"
+	"strings"
+
+	"github.com/rilldata/rill/runtime/drivers"
+)
+
+// wrapClickhouseCompareTimeDim wraps the comparison AST if it has computed time dimension to have different alias than the time column name.
+// Example, in comparison queries we have expression like this: (date_trunc('day', "TIME_DIM") - INTERVAL (DATEDIFF('day', base_time_start, compare_time_start)) day) AS "TIME_DIM"
+// This does not work correctly in ClickHouse and it just does not subtract the interval from the time dimension and return "TIME_DIM" as it is.
+func (e *Executor) wrapClickhouseCompareTimeDim(ast *AST) {
+	if e.olap.Dialect() != drivers.DialectClickHouse || ast.query.ComparisonTimeRange == nil {
+		return
+	}
+
+	e.wrapClickhouseCompareTimeDimWalk(ast, ast.Root)
+}
+
+func (e *Executor) wrapClickhouseCompareTimeDimWalk(a *AST, n *SelectNode) {
```

Comment:
I think we can just assume that `comparison` node will always be present at the Root node of the AST as `JoinComparisonSelect`, that will simplify things and we  won't need this recursive walk. @begelundmuller what do you think?